GitExtUtils/GitArgumentBuilder.cs

```diff
@@ -115,16 +114,18 @@ public sealed class GitArgumentBuilder : ArgumentBuilder
         private static readonly Regex _commandRegex = new Regex("^[a-z0-9_.-]+$", RegexOptions.Compiled);
 
         private readonly List<GitConfigItem> _configItems;
+        private readonly ArgumentString? _gitArgs;
         private readonly string _command;
 
         /// <summary>
         /// Initialises a new <see cref="GitArgumentBuilder"/> for the given <paramref name="command"/>.
         /// </summary>
         /// <param name="command">The git command this builder is compiling arguments for.</param>
         /// <param name="commandConfiguration">Optional source for default command configuration items. Pass <c>null</c> to use the Git Extensions defaults.</param>
+        /// <param name="gitOptions">Optional arguments that are for the git command.  EX: git --no-optional-locks status </param>
         /// <exception cref="ArgumentNullException"><paramref name="command"/> is <c>null</c>.</exception>
         /// <exception cref="ArgumentException"><paramref name="command"/> is an invalid string.</exception>
-        public GitArgumentBuilder([NotNull] string command, [CanBeNull] GitCommandConfiguration commandConfiguration = null)
+        public GitArgumentBuilder([NotNull] string command, [CanBeNull] GitCommandConfiguration commandConfiguration = null, [CanBeNull] ArgumentString? gitOptions = null)
```

Comment:
I think you can avoid making this nullable, and make the default value `default`.

Reply:
Nope. To allow git options to be optional, I had to allow it.

Reply:
I meant that you can use `ArgumentString gitOptions = default`. It doesn't have to be `ArgumentString?`.

Reply:
fixed